src/write.rs

```diff
@@ -0,0 +1,23 @@
+use std::{
+    ffi::CString,
+    io::{self, BufWriter, Seek, Write},
+    u32,
+};
+
+/// An extension for [std::io::Write]
+pub trait WriteExt: Write {
+    /// Read a little endian u32
+    fn write_u32_le(&mut self, value: u32) -> io::Result<()> {
+        self.write(&u32::to_le_bytes(value))?;
+        Ok(())
+    }
```

Comment:
This ignores the total amount of bytes written. Returning the amount of bytes written would actually be more concise

```suggestion
    fn write_u32_le(&mut self, value: u32) -> io::Result<usize> {
        self.write(&u32::to_le_bytes(value))
    }
```